src/main/java/com/mabl/integration/jenkins/MablStepBuilder.java

```diff
@@ -212,14 +213,18 @@ public void perform(
                 getOutputFileLocation(workspace),
                 getEnvironmentVars(run, listener)
         );
-
+    
+        Executor executor = Executor.currentExecutor();
         ExecutorService executorService = Executors.newSingleThreadExecutor();
         Future<Boolean> runnerFuture = executorService.submit(runner);
         try {
             if (runnerFuture.get(EXECUTION_TIMEOUT_SECONDS, SECONDS)) {
                 run.setResult(Result.SUCCESS);
             } else {
                 run.setResult(Result.FAILURE);
+                if (executor != null) {
```

Comment:
Can you elaborate what this fixes? e.g. were we not properly exiting?

Reply:
Jenkins would continue onto the next step even if the run result was marked as a FAILURE. Adding this in aborts the pipeline and marks the result as a FAILURE so that following steps are not executed.

Reply:
Exactly! Even if the plan failed we were still running the next steps on the pipeline. Now if the plans fail we stop the whole thing unless continueOnPlanFailure is true